src/librustc_trans/trans/datum.rs

```diff
@@ -496,8 +509,10 @@ impl<'tcx> Datum<'tcx, Rvalue> {
 
             ByValue => {
                 lvalue_scratch_datum(
-                    bcx, self.ty, name, scope, self,
+                    bcx, self.ty, name, InitAlloca::Dropped, scope, self,
                     |this, bcx, llval| {
+                        debug!("populate call for Datum::to_lvalue_datum_in_scope \
+                                self.ty={:?}", this.ty);
                         call_lifetime_start(bcx, llval);
```

Comment:
Hmm it seems like in some cases the only option would be to inject the lifetime start into the start of the function itself then, right after the alloca.

Does LLVM react okay if there exist multiple lifetime start